### ISSUE TITLE
[Bug Report] Uncaught SyntaxError: The requested module '/node_modules/.vite/vue.js?v=a348567b' does not provide an export named 'createElementBlock'

### ISSUE THREAD

**Issue by Feiox** [element-plus/element-plus#2907]
### Element Plus version
1.0.2-beta.70 (any)

### Version
OS/Browsers macOS 11.4
node  14.5.0
npm  7.14.0
vite  latest

### 

### Steps to reproduce
```
$ npm init vite@latest
$ cd vite-project
$ npm i
$ npm run dev  # ok
$ npm i -S element-plus@latest 
$ npm run dev  # error
```

### What is Expected?
It works normally in vue/cli

### What is actually happening?

Chrome Browser Devtools :
```
HelloWorld.vue:28 Uncaught SyntaxError: The requested module '/node_modules/.vite/vue.js?v=a348567b' does not provide an export named 'createElementBlock'
```

when I remove it
```
 $ npm uninstall element-plus
```
still
```
HelloWorld.vue:28 Uncaught SyntaxError: The requested module '/node_modules/.vite/vue.js?v=a348567b' does not provide an export named 'createElementBlock'
```

**Comment by foyel** [comment]
同出现这个问题
import { resolveComponent as _resolveComponent, createVNode as _createVNode, openBlock as _openBlock, createElementBlock as _createElementBlock } from "/node_modules/.vite/vue.js?v=fa10de3b
![1628665726472](https://user-images.githubusercontent.com/17547882/128985598-d58cac04-8d38-4fd9-b326-31ec48139c77.jpg)

**Comment by foyel** [comment]
估计升级到vue3.2.1就好了

**Comment by ykw0104** [comment]
I got the similar error, have any solution?

 **WARNING  Compiled with 2 warnings                                                                                                       9:27:09 AM
 warning  in ./src/App.vue?vue&type=template&id=7ba5bd90

"export 'createElementBlock' (imported as '_createElementBlock') was not found in 'vue'

 warning  in ./src/App.vue?vue&type=template&id=7ba5bd90

"export 'createElementVNode' (imported as '_createElementVNode') was not found in 'vue'**

**Comment by Alanscut** [comment]
similar error
` WARNING  Compiled with 2 warnings                                                                                                                                                                                       上午10:59:20

 warning  in ./src/App.vue?vue&type=template&id=7ba5bd90

"export 'createElementBlock' (imported as '_createElementBlock') was not found in 'vue'

 warning  in ./src/App.vue?vue&type=template&id=7ba5bd90

"export 'createElementVNode' (imported as '_createElementVNode') was not found in 'vue'
`

**Comment by Alanscut** [comment]
similar error
with npm run serve ：
` WARNING  Compiled with 2 warnings                                                                                                                                                                                        下午2:57:41

 warning  in ./src/App.vue?vue&type=template&id=7ba5bd90

"export 'createElementBlock' (imported as '_createElementBlock') was not found in 'vue'

 warning  in ./src/App.vue?vue&type=template&id=7ba5bd90

"export 'createElementVNode' (imported as '_createElementVNode') was not found in 'vue'
`
error messge in the console of chrome:
`Uncaught TypeError: Object(...) is not a function
    at eval (App.vue?3dfd:2)
    at Module../node_modules/cache-loader/dist/cjs.js?!./node_modules/babel-loader/lib/index.js!./node_modules/vue-loader-v16/dist/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader-v16/dist/index.js?!./src/App.vue?vue&type=template&id=7ba5bd90 (app.js:962)
    at __webpack_require__ (app.js:849)
    at fn (app.js:151)
    at eval (App.vue?4907:1)
    at Module../src/App.vue?vue&type=template&id=7ba5bd90 (app.js:1077)
    at __webpack_require__ (app.js:849)
    at fn (app.js:151)
    at eval (App.vue?dfb6:1)
    at Module../src/App.vue (app.js:1041)`

**Comment by Alanscut** [comment]
similar error
**with npm run serve ：**
` WARNING Compiled with 2 warnings 下午2:57:41 

warning in ./src/App.vue?vue&type=template&id=7ba5bd90 

"export 'createElementBlock' (imported as '_createElementBlock') was not found in 'vue' 

warning in ./src/App.vue?vue&type=template&id=7ba5bd90 

"export 'createElementVNode' (imported as '_createElementVNode') was not found in 'vue'`

**error messge in the console of chrome:**

`Uncaught TypeError: Object(...) is not a function
at eval (App.vue?3dfd:2)
at Module../node_modules/cache-loader/dist/cjs.js?!./node_modules/babel-loader/lib/index.js!./node_modules/vue-loader-v16/dist/templateLoader.js?!./node_modules/cache-loader/dist/cjs.js?!./node_modules/vue-loader-v16/dist/index.js?!./src/App.vue?vue&type=template&id=7ba5bd90 (app.js:962)
at webpack_require (app.js:849)
at fn (app.js:151)
at eval (App.vue?4907:1)
at Module../src/App.vue?vue&type=template&id=7ba5bd90 (app.js:1077)
at webpack_require (app.js:849)
at fn (app.js:151)
at eval (App.vue?dfb6:1)
at Module../src/App.vue (app.js:1041)`

**Comment by ietwangwei** [comment]
same error.
plz.

**Comment by sctnightcore** [comment]
same .

**Comment by joneqian** [comment]
same .

**Comment by noyidoit** [comment]
same error

**Comment by noyidoit** [comment]
Similar error, and I have made a simple test/reproduce for that:
`vue create test`, using default 3 setting
`npm install element-plus --save`
`npm run serve`
and it shows:

![image](https://user-images.githubusercontent.com/26378591/129320558-8e999d3a-4ded-48d2-834b-215e6dca115e.png)

When I access localhost:8080 using browser:
![image](https://user-images.githubusercontent.com/26378591/129320509-c805476e-2bb9-46be-bf52-e6fd3ec18893.png)

**Comment by flydev-fr** [comment]
@noyidoit what is the exact version of Vue ?  

I had this same issue 10 secondes ago, I updated my `package.json` to use `Vue 3.2.2` and the issue is gone.

**Comment by Alanscut** [comment]
@flydev-fr vue3.0.0   3.1.5  3.1.8 will have this error。
i think maybe it will be error  all version between 3.0.0 and 3.1.8

**Comment by noyidoit** [comment]
@flydev-fr sry I forgot for that, it is 3.1.5

**Comment by clai-wh** [comment]
> Similar error, and I have made a simple test/reproduce for that:
> , using default 3 setting
> 
> 
> and it shows:`vue create test``npm install element-plus --save``npm run serve`
> 
> ![image](https://user-images.githubusercontent.com/26378591/129320558-8e999d3a-4ded-48d2-834b-215e6dca115e.png)
> 
> When I access localhost:8080 using browser:
> ![image](https://user-images.githubusercontent.com/26378591/129320509-c805476e-2bb9-46be-bf52-e6fd3ec18893.png)

> 

I(`3.15`) upgraded it to 3.2+and solved the same problem as you

**Comment by Alanscut** [comment]
@flydev-fr @clai-wh 
how did you use vue:3.2.x with element-plus??
![image](https://user-images.githubusercontent.com/21104054/129506053-6b5bc66c-509c-426b-a33e-248bd8f6d4e3.png)

**Comment by clai-wh** [comment]
> @flydev-fr @clai-wh
> how did you use vue:3.2.x with element-plus??
> ![image](https://user-images.githubusercontent.com/21104054/129506053-6b5bc66c-509c-426b-a33e-248bd8f6d4e3.png)

demand import
![屏幕截图 2021-08-16 111706](https://user-images.githubusercontent.com/52877043/129507131-cc74fd8a-62a4-4d59-862a-7eb9dde39961.png)

**Comment by flydev-fr** [comment]
> @flydev-fr @clai-wh
> how did you use vue:3.2.x with element-plus??

```
"element-plus": "^1.0.2-beta.70",
"vue": "^3.2.2",
"vite": "^2.3.8"
```

**Comment by Alanscut** [comment]
@flydev-fr @clai-wh 
it will has `npm error` because version of vue .
![image](https://user-images.githubusercontent.com/21104054/129523837-a469e44e-0e6a-4283-a2b2-eac365d5a49b.png)

so it must use `--force `

**Comment by ben1289** [comment]
![Snipaste_2021-08-16_23-37-45](https://user-images.githubusercontent.com/42233756/129590299-5dc62efe-54cb-46a6-bca3-a8f8dbe9a257.png)
You need to manually modify the Vue version in package.json to 3.2.2, and then restart NPM install.

**Comment by HuRuilin** [comment]
I solved this problem. The reason for this problem is that the peerDependencies configuration version of element-plus is 3.1.x. After installation, the default installed version of vue will be reduced to 3.1.5, so vue-router lacks corresponding methods. The solution is to update the vue version number to 3.2.4 (the version I use). When updating, there may be version conflicts, so you should use npm install --force

**Comment by YiXingYuFeng** [comment]
<img width="513" alt="image" src="https://user-images.githubusercontent.com/20083346/130549535-91873fb3-053f-4ab0-b7e7-61dc34be9586.png">
<img width="831" alt="image" src="https://user-images.githubusercontent.com/20083346/130549546-599d9f3e-69a3-4320-8f6e-5d94c8ee99ba.png">
same error

**Comment by bran-nie** [comment]
This issue should be closed, because I seem to have found the culprit, which is the package @vue/compiler-sfc.

So it is not a problem with element-plus, although element-plus is still a beta version.

---
I helped my friend solve this problem yesterday. She is exactly the same as you encountered. solution:

1. Delete node_modules/
2. Delete the package-lock.json file
3. Modify the dependency in package.json: "@vue/compiler-sfc": "3.0.4"
4. Reinstall: npm install

This problem is caused by the wrong version of the "@vue/compiler-sfc" package. I haven't seen the specific reason. I am currently using version 3.0.4, and a subsequent version will cause a build error. So I first lock the package version in package.json.

![image](https://user-images.githubusercontent.com/18239301/131432754-fae64fc7-ab9e-4ffa-b8f2-9f2ce09c927a.png)

**Comment by bran-nie** [comment]
> This issue should be closed, because I seem to have found the culprit, which is the package @vue/compiler-sfc.
> 
> So it is not a problem with element-plus, although element-plus is still a beta version.
> 
> I helped my friend solve this problem yesterday. She is exactly the same as you encountered. solution:
> 
> 1. Delete node_modules/
> 2. Delete the package-lock.json file
> 3. Modify the dependency in package.json: "@vue/compiler-sfc": "3.0.4"
> 4. Reinstall: npm install
> 
> This problem is caused by the wrong version of the "@vue/compiler-sfc" package. I haven't seen the specific reason. I am currently using version 3.0.4, and a subsequent version will cause a build error. So I first lock the package version in package.json.
> 
> ![image](https://user-images.githubusercontent.com/18239301/131432754-fae64fc7-ab9e-4ffa-b8f2-9f2ce09c927a.png)

---
I tried it and the version of @vue/compiler-sfc before 3.2.0 can be built normally. Compared with the @3.2.6 version of the brand new vue 3 project, it seems that I have not found why compiler-sfc will not work after the upgrade.

**Comment by bran-nie** [comment]
I am ashamed of my questioning @vue/compiler-sfc. In fact, I believe that this time I found the problem.
It should be a project used by a friend, which applied webpack's DLL technology, but because the package version was not locked, the DLL file was not updated after the node_modules package was updated, resulting in incorrect package index pointing. There may be other factors. In short, it's too delicious.
